src/router_core/route_control.c

```diff
@@ -441,6 +454,9 @@ void qdr_route_del_link_route_CT(qdr_core_t *core, qdr_link_route_t *lr)
         }
     }
 
+    if (lr->hash_handle)
+        qd_hash_remove_by_handle(core->addr_lr_al_hash, lr->hash_handle);
```

Comment:
Do we need to keep the hash handle around after this?
If not consider freeing it now and zero the lr->hash_handle.